src/OrchardCore/OrchardCore.Abstractions/Shell/Builders/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -36,21 +38,38 @@ public static IServiceCollection Configure<TOptions>(
     /// Registers an <see cref="IAsyncConfigureOptions{TOptions}"/> used to configure
     /// asynchronously a type of options just after a tenant container is created.
     /// </summary>
-    public static IServiceCollection Configure<TOptions, TConfigure>(this IServiceCollection services)
-        where TOptions : class, IAsyncOptions, new() where TConfigure : IAsyncConfigureOptions<TOptions>
+    public static IServiceCollection Configure<TOptions, TConfigure>(this IServiceCollection services, ILogger logger = null)
```

Comment:
FYI: adding optional parameters to a method is always a binary breaking change: if a library was compiled against the current OC 2.1.x version and calls this method, it will break badly.

(also, taking an `ILogger` parameter in an `IServiceCollection` extension is unusual and looks quite horrible IMHO 😄)

Reply:
Yes. This method is highly unlikely used by anyone. It's not even documented anywhere. But I agree with you it is a binary breaking change. But quite frankly, it does not matter because now with YesSql 5.4.1 merged, libraries will have to recompile and ship new packages since it's all breaking.

About the logger, yes it not a great idea. I just needed the logs in attempt to understand what was happening with that issue.  But, I'll remove it